packages/bp/src/core/bots/bot-service.ts

```diff
@@ -266,10 +264,11 @@ export class BotService {
           .warn(`The bot ${botId} already exists, files in the archive will overwrite existing ones`)
       }
     }
-    const tmpDir = tmp.dirSync({ unsafeCleanup: true })
-    const tmpFolder = tmpDir.name
 
+    let tmpDir: tmp.SynchrounousResult | null = null
     try {
+      tmpDir = tmp.dirSync({ unsafeCleanup: true })
+      const tmpFolder = tmpDir.name
```

Comment:
We use that pattern of tmp.dirSync in like 40 other places (outside the try catch block) however I've never encountered an issue with that method, I guess exceptions are thrown in rare cases

Reply:
Is this just an observation or a request to change the other occurrences as well?

Just for information:
![Screenshot from 2021-09-28 15-04-10](https://user-images.githubusercontent.com/30733005/135149842-b42c1eac-9cd3-4133-901c-5ca5424e78d3.png)